File: specification/specs/query_dsl/span/SpanQuery.ts
class SpanQuery extends QueryBase {
  span_containing?: SpanContainingQuery
  field_masking_span?: SpanFieldMaskingQuery
  span_first?: SpanFirstQuery
  span_gap?: SpanGapQuery
  span_multi?: SpanMultiTermQuery
  span_near?: SpanNearQuery
  span_not?: SpanNotQuery
  span_or?: SpanOrQuery
  span_term?: SpanTermQuery
  span_within?: SpanWithinQuery
}

File: output/csharp/QueryDsl/Span/SpanQuery.cs
using Nest.QueryDsl;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.QueryDsl {

	public class SpanQuery  {
		
		[DataMember(Name="span_containing")]
		public SpanContainingQuery SpanContaining { get; set; }


		[DataMember(Name="field_masking_span")]
		public SpanFieldMaskingQuery FieldMaskingSpan { get; set; }


		[DataMember(Name="span_first")]
		public SpanFirstQuery SpanFirst { get; set; }


		[DataMember(Name="span_gap")]
		public SpanGapQuery SpanGap { get; set; }


		[DataMember(Name="span_multi")]
		public SpanMultiTermQuery SpanMulti { get; set; }


		[DataMember(Name="span_near")]
		public SpanNearQuery SpanNear { get; set; }


		[DataMember(Name="span_not")]
		public SpanNotQuery SpanNot { get; set; }


		[DataMember(Name="span_or")]
		public SpanOrQuery SpanOr { get; set; }


		[DataMember(Name="span_term")]
		public SpanTermQuery SpanTerm { get; set; }


		[DataMember(Name="span_within")]
		public SpanWithinQuery SpanWithin { get; set; }

	}
}
